src/controllers/portfolio/portfolio.ts

```diff
@@ -664,6 +720,87 @@ export class PortfolioController extends EventEmitter {
     this.emitUpdate()
   }
 
+  /**
+   * Learn new tokens from humanizer and debug_traceCall
+   */
+  async learnTokens(tokens: string[], networkId: NetworkId, accountId: AccountId) {
+    const storagePreviousHints = await this.#storage.get('previousHints', {})
+
+    const learnedTokens = storagePreviousHints.learnedTokens || {}
+    const networkLearnedTokens = learnedTokens[networkId] || {}
+
+    const network = this.#settings.networks.find((x) => x.id === networkId)
+
+    if (!network) throw new Error('network not found')
+
+    const portfolioLib = this.initializePortfolioLibIfNeeded(accountId, networkId, network)
+
+    // Get the limit for the network
+    const limit = portfolioLib.deploylessTokens.isLimitedAt24kbData
+      ? LIMITS.deploylessProxyMode
+      : LIMITS.deploylessStateOverrideMode
+
+    // Reached network erc20 limit
+    if (limit.erc20 - Object.keys(networkLearnedTokens).length === THRESHOLD) {
+      await this.cleanLearnedTokens(networkId, limit.erc20, storagePreviousHints)
+    }
+    tokens.forEach((address) => {
+      if (address === ZeroAddress) return
+      if (!(address in networkLearnedTokens)) {
+        networkLearnedTokens[address] = null
+      }
+    })
+    const updatedPreviousHintsStorage = { ...storagePreviousHints }
+    updatedPreviousHintsStorage.learnedTokens[networkId] = networkLearnedTokens
+
+    await this.#storage.set('previousHints', updatedPreviousHintsStorage)
+  }
+
+  #calculateTokensToDeleteCount(networkLearnedTokens: any, limit: number) {
```

Comment:
this sounds excessive, why do we do this?

Reply:
I think we should just always append tokens to this list, and AFTER that remove the oldest. That way we won't need such a complicated mechanism